lib/humanized_id/humanizer.rb

```diff
@@ -0,0 +1,43 @@
+require 'facets'
+
+module HumanizedId
+  class Humanizer
+    def initialize(id:, length: nil, prefix: '')
+      @id               = id
+      @length           = length || id.to_s.length
+      @prefix           = prefix
+      @target_charset   = HumanizedId::CHARACTERSET
+    end
+
+    def generate_humanized_id
+      new_id = translate_id_to_charset
+      new_id = resize(id: new_id)
+      "#{@prefix}#{new_id}"
+    end
+
+    private
+
+    def translate_id_to_charset(id: @id, target_charset: @target_charset)
```

Comment:
Agreed - the action on the variables aren't destructive. This was more of a nicety I gained by using named parameters. It allows us to use either dependency injection or default to instance variables. Not so useful for this method, but I use it in the next method pretty effectively (using the 'new_id' as a dependency injection)